chore(driver): check minimum kernel versions for each driver. (#870)
falcosecurity/libs: driver/CMakeLists.txt

	message(WARNING "Target architecture not officially supported by our drivers!")
endif()

# Load current kernel version

> execute_process(COMMAND uname -r OUTPUT_VARIABLE UNAME_RESULT OUTPUT_STRIP_TRAILING_WHITESPACE)

Author: > Would we want to support extracting uname -r also when building in containers passing the extracted headers,

I think this is rather hard to achieve! I think the best-effort solution is to only `WARNING` when `uname -r` is lower than required kver.

> execute_process(COMMAND uname -r OUTPUT_VARIABLE UNAME_RESULT OUTPUT_STRIP_TRAILING_WHITESPACE)

Reviewer: yeah the warning makes a lot of sense, it says that you are building the driver on a machine that cannot run it, btw there could be corner cases where you build the driver on a not supported machine and you will run it on a supported one, so the warning should be ok!

string(REGEX MATCH "[0-9]+.[0-9]+" LINUX_KERNEL_VERSION ${UNAME_RESULT})
message(STATUS  "Kernel version: ${UNAME_RESULT}")

# Check minimum kernel version
set(kmod_min_kver_map_x86_64 2.6)
set(kmod_min_kver_map_aarch64 3.16)
set(kmod_min_kver_map_s390x 2.6)
if (LINUX_KERNEL_VERSION VERSION_LESS ${kmod_min_kver_map_${TARGET_ARCH}})

> 	message(WARNING "[KMOD] Linux kernel version should be >= ${kmod_min_kver_map_${TARGET_ARCH}} but actual kernel version is: ${UNAME_RESULT}")
> endif()

Reviewer: ```suggestion
	message(WARNING "[KMOD] To run this driver you need a Linux kernel version >= ${kmod_min_kver_map_${TARGET_ARCH}} but actual kernel version is: ${UNAME_RESULT}")
endif()
```
